### PR TITLE
Tasks 1-10: Added

### PR DESCRIPTION
**Pull request by IharLiauchuk**

### CODE REVIEW

File: delivery/asteroidControllerApi.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    
+        const asteroidResponseDto = await getAsteroid(date, showDangerous, showCount);
+    
+        res.send(asteroidResponseDto)
+    } catch (error) {
+        next(new Exception(error.code, error.message));
+    }    
+}
+
+export default  getAsteroids;
```
Comment (mdziarkach) [inline delivery/asteroidControllerApi.js:19]: looks like you duplicated this file

File: usecases/fetchAsteroid.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const result = {};
+    let wereDangerousMeteors = false;
+    const asteroids = [];
+
+    Object.values(jsonBody.near_earth_objects).flatMap(d => d)
+        .forEach(asteroid => {
+            if (!wereDangerousMeteors && asteroid.is_potentially_hazardous_asteroid) {
+                wereDangerousMeteors = true;
+            }
+            asteroids.push({
```
Comment (mdziarkach) [inline usecases/fetchAsteroid.js]: try to implement it without modification data

File: delivery/asteroidControllerApi.js (hunk 1 above)
Comment (IharLiauchuk) [reply delivery/asteroidControllerApi.js:19]: They are the same except line 13. One returns json, the other a page.

File: usecases/fetchAsteroid.js (hunk 2 above)
Comment (IharLiauchuk) [reply usecases/fetchAsteroid.js]: What do you mean by "without modification data"?

File: usecases/fetchAsteroid.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                "id" : asteroid.id,
+                "name" : asteroid.name,
+                "diameter" : asteroid.estimated_diameter.meters,
+                "is_potentially_hazardous_asteroid" : asteroid.is_potentially_hazardous_asteroid,
+                "close_approach_date_full" : asteroid.close_approach_data[0].close_approach_date_full,
+                "relative_velocity" : asteroid.close_approach_data[0].relative_velocity.kilometers_per_second
+            });
+        });
```
Comment (mdziarkach) [inline usecases/fetchAsteroid.js:28]:
```js
const asteroids = Object.values(jsonBody.near_earth_objects).flatMap(d => d)
        .map(asteroid => {
            if (!wereDangerousMeteors && asteroid.is_potentially_hazardous_asteroid) {
                wereDangerousMeteors = true;
            }
            return {
                "id" : asteroid.id,
                "name" : asteroid.name,
                "diameter" : asteroid.estimated_diameter.meters,
                "is_potentially_hazardous_asteroid" : asteroid.is_potentially_hazardous_asteroid,
                "close_approach_date_full" : asteroid.close_approach_data[0].close_approach_date_full,
                "relative_velocity" : asteroid.close_approach_data[0].relative_velocity.kilometers_per_second
            };
        });
```

File: usecases/fetchAsteroid.js (hunk 2 above)
Comment (mdziarkach) [reply usecases/fetchAsteroid.js]: I mean better to return new list without pushing data into asteroids

**Review by mdziarkach — APPROVED** [review]

**Pull request merged**